kafka/producer/main.go

```diff
@@ -29,139 +31,106 @@ const eventHubNamespace = "EVENTHUB_NAMESPACE"
-				}
-
-				// This batch is full - we can send it and create a new one and continue
-				// packaging and sending events.
-				if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
-					log.Printf("Event sending failed %s", err.Error())
-				}
-
-				// create the next batch we'll use for events, ensuring that we use the same options
-				// each time so all the messages go the same target.
-				tmpBatch, err := producerClient.NewEventDataBatch(context.Background(), newBatchOptions)
-
-				if err != nil {
-					log.Printf("Creating new batch failed: %s", err.Error())
-				}
-
-				batch = tmpBatch
-
-				// rewind so we can retry adding this event to a batch
-				i--
-			} else if err != nil {
-				log.Printf("Errored while adding events to batch: %s", err.Error())
-			}
+		err = batch.AddEventData(event, nil)
+		if err != nil {
+			log.Printf("Adding event data to batch failed: %s", err.Error())
 		}
 
-		// if we have any events in the last batch, send it
-		if batch.NumEvents() > 0 {
-			log.Println("sending events")
-			if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
-				log.Printf("Error sending events: %s", err.Error())
-			}
+		if err := producerClient.SendEventDataBatch(context.Background(), batch, nil); err != nil {
+			log.Printf("Event sending failed %s", err.Error())
 		}
-		log.Println("Rest for 10 seconds before sending additional events.")
+
+		select {
+		case sig := <-signals:
+			log.Printf("Got signal: %v", sig)
+			return
+		default:
+		}
+
 		time.Sleep(time.Second * 1)
 	}
 }
 
-func createEventsForDemo() []*azeventhubs.EventData {
-	rand.Seed(time.Now().UnixNano())
-	randomId := rand.Intn(90000) + 10000
-	rawMessage := getEnv(msg)
-	value := fmt.Sprintf("Message Id %d: %s", randomId, rawMessage)
-	encryptedValue, err := encryptMessage(value, "path_to_public_key.pem")
+func createEventsForDemo() *azeventhubs.EventData {
+	eventId += 1
+	rawMessage := util.GetEnv(msg)
+	value := fmt.Sprintf("Message Id %d: %s", eventId, rawMessage)
+	encryptedValue, err := encryptMessage(value)
 	if err != nil {
-		log.Println("Encrypting message failed.")
-	}
-	return []*azeventhubs.EventData{
-		{
-			Body: []byte(encryptedValue),
-		},
-		{
-			Body: []byte(encryptedValue),
-		},
+		log.Printf("Encrypting message failed: %s", err.Error())
 	}
+	return &azeventhubs.EventData{Body: []byte(encryptedValue)}
 }
 
-func encryptMessage(plaintext string, publicKeyPath string) (string, error) {
+func encryptMessage(plaintext string) (string, error) {
 	var pubpem []byte
 	var err error
-	if pkey := getEnv("PUBKEY"); len(pkey) > 0 {
+	if pkey := util.GetEnv("PUBKEY"); len(pkey) > 0 {
 		pubpem = []byte(pkey)
 	}
-	if len(pubpem) == 0 {
-		pubpem, err = os.ReadFile(publicKeyPath)
-		if err != nil {
-			return "", fmt.Errorf("failed to read public key file %v", publicKeyPath)
-		}
-	}
 	block, _ := pem.Decode([]byte(pubpem))
```

Comment:
does this need error handling?

Reply:
Ok. Is `block` guaranteed to not be `nil`?

Reply:
I have not encountered a case where the block is non nil. I would think the ParsePKIPublicKey would catch error if anything goes wrong.